### PR TITLE
feat(icons): redesign brain and related icons

### PR DESCRIPTION
**Pull request by karsa-mistmere**
## What is the purpose of this pull request?

- [x] Icon redesign

### Description
The current brain icon is a bit too detailed in some places but not detailed enough in others and kind of doesn't really look like a brain at all. This PR updates it to look a bit more stylised and at the same time somewhat more organic.

### Alternative icon designs 
image

### Author, credits & license

- [x] The icons are solely my own creation.
- [x] I've based them on the following Lucide icons: the same ones I originally created in #967

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to two points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by jguddas** [comment]
I never thought I would say that, but I think it feels to fun and cute, maybe we could decutify it?

File: icons/brain.svg
Comment (ericfennis) [inline icons/brain.svg:1]: On 100% scale, the center curved line looks a bit like a triangle. Can we try one with a longer line?

Comment (karsa-mistmere) [reply icons/brain.svg:1]: I've updated the PR, let me know what you think.

Comment (jguddas) [reply icons/brain.svg:1]: Feels better to me with the longer line.

**Review by ericfennis — APPROVED** [review]

**Pull request merged**